src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperResolver.cs

```diff
@@ -82,42 +83,124 @@ internal class DefaultTagHelperResolver : ITagHelperResolver
+                if (tagHelper.Documentation != null)
+                {
+                    goto AddDocumentedTagHelper;
+                }
+
+                var typeName = tagHelper.Metadata[ITagHelperDescriptorBuilder.TypeNameKey];
+                var symbol = compilation.GetTypeByMetadataName(typeName);
                 if (symbol != null)
                 {
+                    var tagHelperBuilder = ShallowCopy(typeName, tagHelper);
                     var xml = symbol.GetDocumentationCommentXml();
+
                     if (!string.IsNullOrEmpty(xml))
                     {
-                        var documentation = new XmlMemberDocumentation(xml);
-                        tagHelper.DesignTimeDescriptor.Summary = documentation.GetSummary();
-                        tagHelper.DesignTimeDescriptor.Remarks = documentation.GetRemarks();
+                        tagHelperBuilder.Documentation(xml);
                     }
 
-                    foreach (var attribute in tagHelper.Attributes)
+                    foreach (var attribute in tagHelper.BoundAttributes)
                     {
-                        attribute.DesignTimeDescriptor = attribute.DesignTimeDescriptor ?? new TagHelperAttributeDesignTimeDescriptor();
+                        var propertyName = attribute.Metadata[ITagHelperBoundAttributeDescriptorBuilder.PropertyNameKey];
 
-                        var attributeSymbol = symbol.GetMembers(attribute.PropertyName).FirstOrDefault();
+                        var resolvedAttribute = attribute;
+                        var attributeSymbol = symbol.GetMembers(propertyName).FirstOrDefault();
                         if (attributeSymbol != null)
                         {
                             xml = attributeSymbol.GetDocumentationCommentXml();
                             if (!string.IsNullOrEmpty(xml))
                             {
-                                var documentation = new XmlMemberDocumentation(xml);
-                                attribute.DesignTimeDescriptor.Summary = documentation.GetSummary();
-                                attribute.DesignTimeDescriptor.Remarks = documentation.GetRemarks();
+                                var attributeBuilder = ShallowCopy(typeName, resolvedAttribute);
+                                attributeBuilder.Documentation(xml);
+                                resolvedAttribute = attributeBuilder.Build();
                             }
                         }
+
+                        tagHelperBuilder.BindAttribute(resolvedAttribute);
                     }
+
+                    tagHelper = tagHelperBuilder.Build();
+                }
+
+                AddDocumentedTagHelper:
+
+                documentedTagHelpers.Add(tagHelper);
+            }
+
+            return documentedTagHelpers;
+        }
+
+        private ITagHelperBoundAttributeDescriptorBuilder ShallowCopy(string tagHelperTypeName, BoundAttributeDescriptor attribute)
```

Comment:
Thought about doing that and then realized this was the only part of the API that'd use it. Decided against it until we need it.